src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -563,26 +603,12 @@
         return ! isTrackingRequest;
       },
       code: function(X) {
-        var rejectedApprovalRequest = this.clone();
-        rejectedApprovalRequest.status = this.ApprovalStatus.REJECTED;
-
-        X.approvalRequestDAO.put(rejectedApprovalRequest).then(o => {
-          X.approvalRequestDAO.cmd(this.AbstractDAO.RESET_CMD);
-          this.finished.pub();
-          X.ctrl.add(this.NotificationMessage.create({
-            message: this.SUCCESS_REJECTED
-          }));
-
-          if ( X.currentMenu.id !== X.stack.top[2] ) {
-            X.stack.back();
-          }
-        }, e => {
-          this.throwError.pub(e);
-          X.ctrl.add(this.NotificationMessage.create({
-            message: e.message,
-            type: 'error'
-          }));
-        });
+        this.objectSummaryView.add(this.Popup.create({ backgroundColor: 'transparent' }).tag({
+          class: "foam.u2.MemoModal",
+          onExecute: this.rejectWithMemo.bind(this),
+          isMemoRequired: true,
+          title: "Add a memo to this (required)"
```

Comment:
This does not read well.  Perhaps
'Please provide a reason'

Reply:
Done.